client/src/sections/material/EmbedLinkForm.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { Fragment } from 'react';
+import { type SubmitHandler, useForm } from 'react-hook-form';
+import type { IsActiveProps, MaterialFormInput } from '@/src/shared/utils';
+import RFInputField from '@/src/shared/components/ReactForm/RFInputField';
+import DivCollapse from '@/src/shared/components/Collapse/DivCollapse';
+import RFTextField from '@/src/shared/components/ReactForm/RFTextField';
+import RFSelectField from '@/src/shared/components/ReactForm/RFSelectField';
+import Button from '@/src/shared/components/Button';
+
+const EmbedLinkForm: React.FunctionComponent<IsActiveProps> = ({
+  isActive
+}: IsActiveProps) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<MaterialFormInput>();
+
+  const onSubmit: SubmitHandler<MaterialFormInput> = async (
+    data: MaterialFormInput
+  ): Promise<void> => {
+    console.log(data);
+  };
+
+  const folder = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Folder 1' },
+    { value: '2', label: 'Example Folder 2' },
+    { value: '3', label: 'Example Folder 3' }
+  ];
+
+  const categories = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Category 1' },
+    { value: '2', label: 'Example Category 2' },
+    { value: '3', label: 'Example Category 3' }
+  ];
+
+  const coureses = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Course 1' },
+    { value: '2', label: 'Example Course 2' },
+    { value: '3', label: 'Example Course 3' }
+  ];
+
+  return (
+    <Fragment>
+      <div
+        className={`p-4 max-h-[800px] overflow-y-scroll ${
+          isActive ? 'block' : 'hidden'
+        }`}
+      >
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <h2 className="text-md text-blue-900 mb-5">Embed a Web Page</h2>
+
+          <RFInputField
+            label="Name"
+            {...register('name', { required: true, minLength: 5 })}
+            error={
+              errors.name !== undefined &&
```

Comment:
Apply to all related

```suggestion
              !!errors.name &&
```

Reply:
in order for me to implement this, I had to add a new ESLint rule `"@typescript-eslint/strict-boolean-expressions": "off",`